spring-integration-test/src/main/java/org/springframework/integration/test/context/MockIntegrationContext.java

```diff
@@ -108,9 +114,19 @@ else if (endpoint instanceof ReactiveStreamsConsumer) {
 					else if (endpoint instanceof IntegrationConsumer) {
 						directFieldAccessor.setPropertyValue(HANDLER, e.getValue());
 					}
+					if (lifecycle != null && lifecycle.isAutoStartup()) {
```

Comment:
Doesn't this have potential to break existing tests?

Maybe this just matters for constructed examples. But I could imagine something like this:
```
@BeforeClass
void init() {
  nonAutostartupHandler.start();
}

@After 
void cleanup() {
  mockIntegrationContext.reset()
}

@Test
void t1() {
  // do something with handler
}

@Test 
void t2() {
  // expect handler to be still running
}
```

Admittedly the `@BeforeClass` is somewhat artificial. But I could see a  handler being manually started as part of context initialization.

I'm sorry for not contributing a fix myself, I had a pretty busy week at work.

Reply:
`nonAutostartupHandler.start();` - you definitely start it manually. The framework can't assume when and how you have started it or going to.
Only the way for the framework to start it when you have that `autoStartup = true`.

I admit that it may break some peculiar test, but at the same time the test is wrong because it has been relied on bug which we are fixing here.

On the other hand I don't worry too much about breaking changes for tests. That's not that part of project which may make your customer unhappy.